.github/workflows/e2e-tests.yaml

```diff
@@ -0,0 +1,66 @@
+name: End-to-End Testing
+
+on:
+  push:
+    # NOTE(antoineco): To avoid exposing repository secrets to user supplied
+    # code in pull requests, we only allow this workflow to run on the 'main'
+    # branch, where all code changes are expected to have been reviewed by
+    # maintainers.
+    #
+    # We may want to relax this in the future, on condition that safeguards are
+    # added to prevent the workflow from running automatically when an external
+    # contributor submits code changes inside the .github/workflows/ or
+    # test/e2e/ directories.
+    #
+    # Ref. https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+    branches: [ main ]
```

Comment:
Adding a note for myself about recommended security practices in GitHub Actions: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions